connman/src/firewall-iptables.c

```diff
@@ -2316,82 +2317,40 @@ static int init_dynamic_firewall_rules(const char *file)
 
 static int init_all_dynamic_firewall_rules(void)
 {
-	GList *iter;
-	GError *error = NULL;
-	GDir *dir;
-	const char *filename = NULL;
-	char *filepath = NULL;
 	int err;
 
 	err = init_dynamic_firewall_rules(FIREWALLCONFIGFILE);
-
-	if (g_file_test(FIREWALLCONFIGDIR, G_FILE_TEST_IS_DIR)) {
-		dir = g_dir_open(FIREWALLCONFIGDIR, 0, &error);
-
-		if (!dir) {
-			if (error) {
-				DBG("cannot open dir, error: %s",
-							error->message);
-				g_clear_error(&error);
-			}
-			goto out;
-		}
-
-		DBG("read configs from %s", FIREWALLCONFIGDIR);
-
-		/*
-		 * Ordering of files is not guaranteed with g_dir_open(). Read
-		 * the filenames into sorted GList.
-		 */
-		while ((filename = g_dir_read_name(dir))) {
-			/* Read configs that have firewall.conf suffix */
-			if (!g_str_has_suffix(filename, FIREWALLFILE))
-				continue;
-
-			/*
-			 * Prepend read files into list of configuration
-			 * files to be used in checks when new configurations
-			 * are added to avoid unnecessary reads of already read
-			 * configurations. Sort list after all are added.
-			 */
-			configuration_files = g_list_prepend(
-						configuration_files,
-						g_strdup(filename));
-		}
-
-		configuration_files = g_list_sort(configuration_files,
-					(GCompareFunc)g_strcmp0);
-
-		for (iter = configuration_files; iter; iter = iter->next) {
-			filename = iter->data;
-
-			filepath = g_strconcat(FIREWALLCONFIGDIR, filename,
-						NULL);
-			DBG("reading config %s", filepath);
-
-			/* Allow also symbolic links in configs */
-			if (g_file_test(filepath, G_FILE_TEST_IS_REGULAR)) {
-				if (init_dynamic_firewall_rules(filepath))
-					DBG("invalid firewall config");
-			}
-
-			g_free(filepath);
-		}
-
-		g_dir_close(dir);
-	} else {
-		DBG("no config dir %s", FIREWALLCONFIGDIR);
+	if (err) {
+		connman_warn("Failed to read main firewall config %s",
+							FIREWALLCONFIGFILE);
+		return err;
 	}
 
-	/* Error loading main configuration */
-	if (err)
+	err = util_read_config_files_from(FIREWALLCONFIGDIR, FIREWALLFILE,
```

Comment:
The config file suffix used here is `FIREWALLFILE` which expands to `firewall.conf`. Same in line 2677 too. Should these be `CONFIGSUFFIX` instead for `.conf` suffix?

Reply:
It was an old idea to avoid them getting mixed with normal configs, but they are in different directories now. So it was intentional choice back then and would need config renaming now. Maybe we keep it? It was documented I think.

Reply:
I looked at the config files on some device of mine and noticed that it's intentional, I was thinking it was a typo or something. Let's keep it like this, although the suffix is getting redundant as a side effect.